core/templates/dev/head/pages/learner_dashboard/LearnerDashboard.js

```diff
@@ -221,79 +238,83 @@ oppia.controller('LearnerDashboard', [
       // so that special cases can be handled while sorting subscriptions.
       var value = entity[$scope.currentSubscribersSortType];
       if ($scope.currentSubscribersSortType ===
-          SUBSCRIPTION_SORT_BY_KEYS.IMPACT) {
+          SUBSCRIPTION_SORT_BY_KEYS_AND_I18N_IDS.IMPACT.key) {
         value = (value || 0);
       }
       return value;
     };
 
-    $scope.sortFeedbackFunction = function(entity) {
-      return entity[$scope.currentFeedbackThreadsSortType];
-    }
+    $scope.sortFeedbackThreadsFunction = function(feedbackThread) {
+      return feedbackThread[$scope.currentFeedbackThreadsSortType];
+    };
 
     $scope.onClickThread = function(
-      explorationId, threadId, explorationTitle) {
-      var threadDataUrl = (
-        '/learnerdashboardthreadhandler/' + explorationId + '/' + threadId);
-      $scope.threadTitle = explorationTitle;
+      threadStatus, explorationId, threadId, explorationTitle) {
+      var threadDataUrl = UrlInterpolationService.interpolateUrl(
+        '/learnerdashboardthreadhandler/<explorationId>/<threadId>', {
+          explorationId: explorationId,
+          threadId: threadId
+        });
+      $scope.explorationTitle = explorationTitle;
       $scope.feedbackThreadActive = true;
+      $scope.threadStatus = threadStatus;
       $scope.explorationId = explorationId;
       $scope.threadId = threadId;
 
       for (var index = 0; index < $scope.threadSummaries.length; index++) {
-        if ($scope.threadSummaries[index].exploration_id === explorationId &&
-            $scope.threadSummaries[index].thread_id === threadId) {
-          $scope.threadIndex = index;
-          var threadSummary = $scope.threadSummaries[index]
-          if (typeof threadSummary.second_last_message_read !== 'undefined') {
-            $scope.threadSummaries[index].second_last_message_read = true;
+        if ($scope.threadSummaries[index].explorationId === explorationId &&
+            $scope.threadSummaries[index].threadId === threadId) {
+          threadIndex = index;
+          var threadSummary = $scope.threadSummaries[index];
+          if (threadSummary.authorSecondLastMessage) {
```

Comment:
All this stuff should be part of the domain object -- something like threadSummary.markAllMessagesAsRead(). (Also, since you've gone to the trouble of defining threadSummary, you might as well use it throughout.)